odoo/models.py

```diff
@@ -5183,11 +5277,45 @@ def _parent_store_update(self):
             self.env.cache.update(records, self._fields['parent_path'], updated.values())
             records.modified(['parent_path'])
 
+    def _clean_properties(self):
```

Comment:
Add documentation: "remove property values that are no longer in the definition" - or something similar.